Use 'Withdrawal' and 'Deducted' labels when referring to withdrawal deposits, to more accurately communicate the type of transaction that has occurred

Jinksi opened the pull request:
Fixes #7802

#### Changes proposed in this Pull Request

1. The `Paid` deposit status is now `Completed (paid)`
2. When viewing a 'withdrawal' deposit, the following labels are used:
	* `Completed (deducted)` instead of `Paid`
	* `Withdrawal` instead of `Deposit`
	* This applies to the following screens and components:
		* Deposit details screen
		* Deposit list screen
		* Deposit list screen advanced filters
		* Payments → Overview → Recent deposits list
1. When viewing a list of transactions on the deposit details screen, **the label `Deposit transactions` is now simply `Transactions`**, to reduce some confusion that could be introduced by the term `Withdrawal transactions`.

**TODO**
- [x] Add tests to cover the "Withdrawal" rendering
	- [x] Deposits list
	- [x] Deposits list advanced filters
	- [x] Deposit details
	- [x] Recent deposits list
- [x] Add `Type` to recent deposits list **reverted, redundant with the `Completed (deducted)` label**
- [x] Change `Paid` and `Deducted` to `Completed (paid)` and `Completed (deducted)`
- [x] Re-capture screenshots of changes
- [x] Clarify all changes in PR description above
- [x] Fix line-wrapping issue with the deposit details status label
	- Fixed in #8382 

**Deposit details screen:**

Withdrawal:

<img width="1403" alt="image" src="https://github.com/user-attachments/assets/36e5f05b-e5a5-4dc1-bbd1-8bf3208e65d6">

Deposit:

<img width="1398" alt="Screenshot 2024-10-21 at 14 27 58" src="https://github.com/user-attachments/assets/5a27d616-910a-4ff1-85b6-b43510e0291c">


**Deposit list screen:**
<img width="1398" alt="Screenshot 2024-10-21 at 14 28 09" src="https://github.com/user-attachments/assets/a54630b8-2418-47c3-9b14-07970d05ce63">

**Deposit list screen advanced filters:**

The option for `paid` now shows `Completed`.

<img width="1194" alt="Screenshot 2024-10-21 at 14 28 32" src="https://github.com/user-attachments/assets/f177e876-ae0e-4510-ae6b-e82abbee5119">

**Payments overview recent deposits list:**

![image](https://github.com/user-attachments/assets/885a943e-5c87-4b8d-9d85-de23fa165a45)




#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->


> [!TIP]
> I've put this PR's changes on the [Helix Demo Site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits&paged=3), if you have access. If you go back to [page 3 of the Payments → Deposits list](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits&paged=3), you can see a mix of Withdrawal/Deposits.


* Change to this branch and run `npm run build`
* Hopefully your store has at least one deposit and one withdrawal deposit. If not:
	* You can create withdrawal deposits by creating one or more disputed transactions using the Stripe test card `4000000000000259`
	* Ensure your store has a Daily deposit schedule – **Payments → Settings → Deposit schedule**
	* Then wait around 24h for the withdrawal to occur
* View the Deposit list screen. 
	* On `trunk` withdrawal deposits will appear as type `Deposit` with a `Paid` status
	* On this branch, ensure that withdrawal deposits appear as type `Withdrawal` with a `Deducted` status (see screenshot above)
	* Open advanced filters and choose `Status. The `Paid` option has been re-labelled to `Paid / Deducted`. Note that we don't support filtering by `Deducted`, since it is a display/label only and not a status recognised by the WooPayments Server.
* Ensure the `Deducted` status appears next to withdrawals on the **Payments → Overview** recent deposits list.
* From the deposit list screen, click a **withdrawal** to view the deposit details screen
	* Ensure you see `Withdrawal date`, `Deducted` and `Withdrawal transactions` (see screenshot above)
* From the deposit list screen, click a **deposit** to view the deposit details screen
	* Ensure you see `Deposit date`, `Paid` and `Deposit transactions`


-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. **N/A**
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. **Added to Helix New in WooPayments post draft**

Jinksi committed:
Show paid withdrawals as 'Deducted' in deposit status chip

Jinksi committed:
Show paid withdrawals as 'Deducted' in deposit detail screen status indicator

Jinksi committed:
Fix imported dependency order to improve code quality

Jinksi committed:
Replace 'Deposit' with 'Withdrawal' on deposit details screen

Jinksi committed:
Don't use 'deducted' status for deposit list advanced filters or queries

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9500` or branch name `fix/7802-deposit-withdrawal-text-changes` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7802-deposit-withdrawal-text-changes) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 59b3c3cff102b67e367f7284fb2b4c3fc1c2db50
- Build time: 2024-11-06 00:44:45 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi committed:
Rename `displayStatus` to `depositStatusLabels` to improve clarity

Jinksi committed:
Add changelog entry

Jinksi committed:
Render advanced filter deposit status label as 'Paid / Deducted' to match changes to withdrawal status label

Jinksi commented:
Ready for review – note I'll be adding some additional integration tests to ensure the new withdrawal labels are rendered correctly, as mentioned in the PR description TODO.

Jinksi committed:
Add DepositOverview test case for withdrawal labels

Jinksi committed:
Add a withdrawal test case to the DepositsOverview component tests

Jinksi committed:
Add a withdrawal test case to the deposits list snapshots

Jinksi committed:
Add a withdrawal test case to the transactions list snapshots

Jinksi commented on client/transactions/list/test/__snapshots__/index.tsx.snap:
```diff
@@ -131,15 +131,15 @@ exports[`Transactions list renders correctly when can filter by several currenci
                       <span
                         class="screen-reader-text"
-                        id="search-inline-input-15"
+                        id="search-inline-input-16"
```
These snapshot changes to the rendered HTML `id` numbers can be ignored – this occurs because I've added a new test case being rendered, bumping many of these ids by +1.

Jinksi commented:
PHP/e2e GH actions failing due to error when downloading from `https://develop.svn.wordpress.org/`: see slack thread p1727686155235229-slack-C02AVAR9B.

Jinksi commented:
![image](https://github.com/user-attachments/assets/d19d6257-3eb1-4d0c-a66a-c16e8825afd4)

Apologies for the large diff – rest assured this is not as large as it seems, once you see past the new test snapshots introduced.

nagpai commented:
### Manual test

The changes test well on a manual test 🥇 

* ✅  - Withdrawal deposits appear as type `Withdrawal` with a `Deducted` status

* ✅  - Open advanced filters and choose `Status. The `Paid` option has been re-labelled to `Paid / Deducted`. 
* ✅  - Ensure the `Deducted` status appears next to withdrawals on the **Payments → Overview** recent deposits list.
* ✅  From the deposit list screen, click a **withdrawal** to view the deposit details screen
	* Ensure you see `Withdrawal date`, `Deducted` and `Withdrawal transactions` (see screenshot above)
* ✅  - From the deposit list screen, click a **deposit** to view the deposit details screen
	* Ensure you see `Deposit date`, `Paid` and `Deposit transactions`

nagpai commented on client/deposits/strings.ts:
```diff
@@ -16,8 +16,17 @@ export const displayType = {
+ * 'deducted' represents a deposit of the type 'withdrawal' and status 'paid'.
+ */
+export const depositStatusLabels: Record<
+	DepositStatus | 'deducted',
```
At the first glance I wondered why `deducted` is not added within the `DepositStatus` type. I then realized `deducted` is not a status coming from Stripe, but what we use for labeling. I read the doc comment later too 💡

nagpai reviewed (approved):
I did a round of review to check for any improvements. I could not find any. The code is good to merge as is now. I understand @Jinksi , you will be adding some more tests before that.

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

haszari reviewed (approved):
This all looks good. I briefly reviewed the code (there are lots of low-risk changes, so I didn't go deep).

Reviewing the UX I see some areas we might want to revisit or simplify in this PR. These are my main concerns:

1. New `Deducted` UI-only status muddies the waters.
2. Pervasive changes outside the payout details screen.
1. `Withdrawal transactions` heading for payout details table is confusing, since some transactions are negative and others are positive.

None of these are blockers, but maybe worth more discussion before merge. Also relevant for our V2 design / model, maybe we need to revisit `paid` status to better handle negative payouts.

### (1) New `Deducted` UI-only status muddies the waters

In the payouts (deposits) list screen, we now have a filter for `Paid / Deducted`. The `deducted` status is UI only, it doesn't match our underlying model.

As I see it, the problem here is mixing the "direction" of the payout with the status. IMO this is something we could better handle in V2 / going forward – e.g. `settled` instead of `paid` (and `deducted`). If we want to consider that, we should keep it in mind for any changes we make here – i.e. work towards that.

The real meaning of `paid` status is **`settled`** – the money has finished moving. It's vastly more common that this is the same as `paid`, since most payouts are positive.

Notably [Stripe refers to this as `paid`](https://docs.stripe.com/api/payouts/object#payout_object-status); [we also use `paid` in current V2 schema](https://github.com/Automattic/woocommerce-payments-server/blob/trunk/server/wp-content/lib/woopayments/Domains/Payouts/PayoutStateMachine/Enums/PayoutStateName.php#L14); [Adyen _maybe_ uses `booked`](https://docs.adyen.com/api-explorer/transfers/4/get/transfers/(id)) (? TBC).

### (2) Pervasive changes outside the payout details screen
My reading of #7802 was that we needed to fix the payout detail screen to not say `Paid`. 

However I can see that once we start pulling that string we would want to update the status in list screen, overview etc. 

This is closely related to (1), i.e. we have to simulate this fake status in a bunch of places, and I don't know if that's worthwhile or not. Ideally we'd fix the underlying model rather than patch in multiple places.

### (3) `Withdrawal transactions` heading for payout details table is confusing

This is minor, maybe a blocker. The heading of the table says `Withdrawal transactions` which IMO is more confusing than `Deposit (aka payout) transactions` or just `Transactions`.

This _might_ lead to more support tickets/questions.

## mitigation options for the above concerns
#### (1) and (2) - new UI-only status
- Show `Paid` as before, everywhere except the details screen. I.e. consistent with model everywhere, but clearer and inconsistent on detail screen (the pain point highlighted in #7802). 
- Show `Paid` as before, but add a `Paid (deducted)` label when appropriate. UI status is consistent with model (`paid`), but acknowledges direction explicitly. The filter options don't need to be changed, i.e. `Paid` works for both.
- Use a new `Settled` or `Completed` status (if we plan to do this for V2 – TBD!). This could be presented as `Settled (paid)` and `Settled (deducted)` labels, i.e. transparently combine status and direction.

My preference would be for the last option, if @souravdebnath1986  / @rogermattic like the idea. Otherwise, option 2 seems compatible with current PR but more aligned with underlying model. So maybe option 2 and consider option 3 as follow up?

#### (3) `Withdrawal transactions`
Change this label to `Transactions`, maybe for all "directions", i.e. keep it super simple.

Jinksi commented:
> (3) Withdrawal transactions
> Change this label to Transactions, maybe for all "directions", i.e. keep it super simple.

I agree on this as a preferred option to `Deposit transactions` and `Withdrawal transaction` – I'll implement this change.

**Update** added in cbbf62c30aaa832916935c0b45fe9b1692269f97

Jinksi committed:
Simplify the deposit details heading `Deposit / Withdrawal transactions` to `Transactions`

rogermattic commented:
>(3) Withdrawal transactions
Change this label to Transactions, maybe for all "directions", i.e. keep it super simple.

@Jinksi @haszari  I also agree with labeling all of that `Transactions`. Sorry, it was my bad. I didn't think it through when I suggested the heading in my comment in [#7802](https://github.com/Automattic/woocommerce-payments/issues/7802#issuecomment-1994806792)

Regarding **(1) New Deducted UI-only status muddies the waters**. 

I'm not sure about this option:
>Show Paid as before, but add a Paid (deducted) label when appropriate. UI status is consistent with model (paid), but acknowledges direction explicitly. The filter options don't need to be changed, i.e. Paid works for both.

as for me a `Paid (deducted)` sounds really confusing.

also, 
>Note that we don't support filtering by Deducted, since it is a display/label only and not a status recognised by the WooPayments Server.

ideally, the user should be able to filter by the `Deducted` status if we commit to displaying it.

So, I think option 3 would be ideal:

>Use a new Settled or Completed status (if we plan to do this for V2 – TBD!). This could be presented as Settled (paid) and Settled (deducted) labels, i.e. transparently combine status and direction.

For me, with the relevant `Types` (on the condition that the user still can't hide the `Type` column) – `Deposit` (`Payout`) / `Withdrawal` –  even just a standalone `Settled` is sufficient. However, the `Settled (paid)` and `Settled (deducted)` labels would be a bonus.

Jinksi commented:
> [!TIP]
> I've put this PR's changes on the [Helix Demo Site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits&paged=3), if you have access. If you go back to [page 3 of the Payments → Deposits list](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits&paged=3), you can see a mix of Withdrawal/Deposits.

rogermattic commented:
Thanks for adding the changes to the Helix Demo Site, I really appreciate it – really works the best for me to test this way :)

As I was testing this I had a thought @Jinksi @haszari that maybe a `Completed` label would work best? I'm just worrying that `Settled` may need some extra explanation for some folks? 

Also, an update on my note above:
>For me, with the relevant Types (on the condition that the user still can't hide the Type column) – Deposit (Payout) / Withdrawal – even just a standalone Settled is sufficient. However, the Settled (paid) and Settled (deducted) labels would be a bonus.

I still think `Settled` / `Completed` as a status would be sufficient, but I understand we might need a bit more information to patch this table on the Payments Overview...

<img width="768" alt="Screenshot 2024-10-03 at 17 51 21" src="https://github.com/user-attachments/assets/5d411e8e-45fb-4f6c-b94e-345d5ca89d04">

Would it be possible to bring in the `Type` column to this table? Otherwise we need to add `Settled/Completed (paid)` / `Settled/Completed (deducted)`


WDYT?

Jinksi commented:
> Would it be possible to bring in the `Type` column to this table?

This is easily done, I'll make the change.

**Added in 9f33dbb**

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi committed:
Add "Type" column to payments overview recent deposits list

Jinksi committed:
Update snapshot for deposits overview

Jinksi committed:
Prefer inferred types to simplify code

Jinksi committed:
Change Paid / Deducted to Completed (paid/deducted)

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi commented:
@rogermattic @haszari now that I've implemented the changes we discussed, could you please take a look at the screenshots above (or the [Helix demo site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits&paged=3)) and share your thoughts on the changes:

* Added the `Type` column to the Payments → Overview → Deposits card
* On the deposit details screen, `Deposit transactions` is now `Transactions`
* `Paid`  is now `Completed (paid)`
* `Deducted` is now `Completed (deducted)`

There will be some followup CSS alignment issues to correct, I'll do once we confirm we're happy with the naming above – e.g. in the `Type` column on the Payments → Overview → Deposits card.

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

haszari reviewed (approved):
This all looks great! Nice to do this holistically, details matter!

I tested by reviewing the Helix demo site – I haven't reviewed code (assuming this is 👌🏻  at this point).

One area might be worth another look – the payouts overview table.

> Would it be possible to bring in the Type column to this table? Otherwise we need to add Settled/Completed (paid) / Settled/Completed (deducted)

This seems redundant to me, so adds clutter (especially since withdrawals are a rare edge case). @Jinksi will the `Status` show `Completed (deducted)` for (rare) withdrawals? If so, the `Type` might not be needed.

<img width="756" alt="Screenshot 2024-11-05 at 1 36 25 PM" src="https://github.com/user-attachments/assets/fe4a6f6b-5c30-4bff-82f8-288d07382123">

Jinksi commented:
> @Jinksi will the Status show `Completed (deducted)` for (rare) withdrawals? If so, the Type might not be needed.

Yes, the status in the recent deposits list will show `Completed (deducted)`, therefore I agree that the `Type` column is redundant here and clutters the UI. I'll remove it.

![image](https://github.com/user-attachments/assets/5076322f-6f91-40a5-9449-e6fc2a213e63)

**Update**: I've reverted the new `Type` column in the recent deposits list in 6106fc1 and e806f82:


![image](https://github.com/user-attachments/assets/7827369f-fa5d-4298-a485-e6ae8ad04aad)

Jinksi committed:
Revert "Add "Type" column to payments overview recent deposits list"

This reverts commit 9f33dbb14213f67a949392307becc1ddf449dc56.

Jinksi committed:
Revert "Update snapshot for deposits overview"

This reverts commit 95d44acbbb6a2b2e2c43ae14c4c46d1fcd01c612.

rogermattic commented:
Just tested it @Jinksi @haszari using the demo site (thanks for this)!

Sorry for the suggestion to add the `Type` column to the overview table. I agree it's redundant when usually this table only shows  `Deposits`. Thanks for making the change and reverting it, sorry for the time wasted :(  

Otherwise everything looks great.  

I only have one extra question – I was wondering why this label wraps if there seems to be enough space for displaying it all in one line?  What's the rule behind it?
<img width="135" alt="Screenshot 2024-11-06 at 00 42 31" src="https://github.com/user-attachments/assets/32e584c7-3b7a-4674-adae-882fc6e11989">

Jinksi committed:
Merge branch 'develop' into fix/7802-deposit-withdrawal-text-changes

Jinksi commented:
> I only have one extra question – I was wondering why this label wraps if there seems to be enough space for displaying it all in one line? What's the rule behind it?

Thanks for catching this @rogermattic 🙏 I'll fix this line-wrapping issue.

Jinksi commented:
It turns out, the deposit details status label was fixed in #8382, which has therefore been fixed once I merged `develop` into this branch.

![image](https://github.com/user-attachments/assets/805863f7-9a9f-45e3-a0b2-135626a5c01a)

rogermattic commented:
Awesome! Thanks @Jinksi !
